anni/src/database/helpers/cpic-constructors.ts

```diff
@@ -78,6 +77,65 @@ function drugFromRecommendation(recommendation: CpicRecommendation): IDrug_Any {
     };
 }
 
+// initially (before contracting) guideline.externalData and values
+// (phenotype descriptions) of guideline.lookupkey should be of length
+// 1, as we set it this way ourselves in guidelineFromRecommendation
+function ensureInitialGuidelineStructure(guideline: IGuideline_Any) {
+    if (
+        guideline.externalData.length != 1 ||
+        !Object.values(guideline.lookupkey).every((value) => value.length == 1)
+    ) {
+        throw Error('Expected different initial guideline data structure');
+    }
+}
+
+// merge same-phenotype guidelines
+function contractByPhenotype(
+    drugsWithGuidelines: Array<DrugWithGuidelines>,
+): Array<DrugWithGuidelines> {
+    return drugsWithGuidelines.map(({ drug, guidelines }) => {
+        const phenotypeMap = new Map<string, IGuideline_Any>();
+        guidelines.forEach((guideline) => {
+            const key = phenotypeKey(guideline);
+            if (phenotypeMap.has(key)) {
+                const existingGuideline = phenotypeMap.get(key);
+                // ensure that we don't miss information when
+                // getting only first index from externalData
+                // and lookupkey[gene]
+                ensureInitialGuidelineStructure(guideline);
+                existingGuideline!.externalData.push(guideline.externalData[0]);
+                Object.keys(existingGuideline!.lookupkey).forEach((gene) => {
+                    existingGuideline!.lookupkey[gene].push(
+                        guideline.lookupkey[gene][0],
+                    );
+                });
```

Comment:
Could avoid `has`-call and bang-operators to unwrap optional `existingGuideline`

```suggestion
            const existingGuideline = phenotypeMap.get(key);
            if (existingGuideline) {
                // ensure that we don't miss information when
                // getting only first index from externalData
                // and lookupkey[gene]
                ensureInitialGuidelineStructure(guideline);
                existingGuideline.externalData.push(guideline.externalData[0]);
                Object.keys(existingGuideline.lookupkey).forEach((gene) => {
                    existingGuideline.lookupkey[gene].push(
                        guideline.lookupkey[gene][0],
                    );
                });
```